backend/src/common/plugins/authorization/authorization.plugin.ts

```diff
@@ -0,0 +1,68 @@
+import { type FastifyInstance, type FastifyPluginCallback } from 'fastify';
+import fp from 'fastify-plugin';
+
+import { type UserFindResponseDto } from '~/bundles/users/types/types.js';
+import { type UserService } from '~/bundles/users/users.js';
+import { ControllerHooks } from '~/common/controller/controller.js';
+import { HttpCode } from '~/common/http/http.js';
+import { tokenService } from '~/common/services/services.js';
+
+type AuthOptions = {
+    services: {
+        userService: UserService;
+    };
+    routesWhiteList: string[];
+};
+
+declare module 'fastify' {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+    interface FastifyRequest {
+        user: UserFindResponseDto;
+    }
+}
+
+const authorizationPlugin: FastifyPluginCallback<AuthOptions> = (
+    fastify: FastifyInstance,
+    { services, routesWhiteList }: AuthOptions,
+    done,
+) => {
+    fastify.decorateRequest('user', null);
+
+    fastify.addHook(ControllerHooks.ON_REQUEST, async (request, reply) => {
+        try {
+            const isWhiteRoute = routesWhiteList.includes(request.routerPath);
```

Comment:
we should check the method
also, we can move it to server-application helper

Reply:
Just to clarify,
You want to make isWhiteRoute a helper and it should also check the request method used?

If so, should I only check for GET methods and only allow such methods?

Reply:
yes. you can add something like `checkWhiteRoute`

All routes should have path and method

Reply:
checkWhiteRoute helper:

![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/90120572/98bdde61-0eeb-4d37-b069-183c2d6aed12)